src/transformers/models/pvt_v2/configuration_pvt_v2.py

```diff
@@ -0,0 +1,167 @@
+            The dropout probability for stochastic depth, used in the blocks of the Transformer encoder.
+        layer_norm_eps (`float`, *optional*, defaults to 1e-06):
+            The epsilon used by the layer normalization layers.
+        qkv_bias (`bool`, *optional*, defaults to `True`):
+            Whether or not a learnable bias should be added to the queries, keys and values.
+        linear_attention (`bool`, *optional*, defaults to `False`):
+            Use linear attention complexity. If set to True, `sr_ratio` is ignored and average pooling is used for
+            dimensionality reduction in the attention layers rather than strided convolution.
+        out_features (`List[str]`, *optional*):
+            If used as backbone, list of features to output. Can be any of `"stem"`, `"stage1"`, `"stage2"`, etc.
+            (depending on how many stages the model has). If unset and `out_indices` is set, will default to the
+            corresponding stages. If unset and `out_indices` is unset, will default to the last stage.
+        out_indices (`List[int]`, *optional*):
+            If used as backbone, list of indices of features to output. Can be any of 0, 1, 2, etc. (depending on how
+            many stages the model has). If unset and `out_features` is set, will default to the corresponding stages.
+            If unset and `out_features` is unset, will default to the last stage.
+    Example:
+
+    ```python
+    >>> from transformers import PvtV2Model, PvtV2Config
+
+    >>> # Initializing a pvt_v2_b0 style configuration
+    >>> configuration = PvtV2Config()
+
+    >>> # Initializing a model from the FoamoftheSea/pvt_v2_b0 style configuration
+    >>> model = PvtV2Model(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "pvt_v2"
+
+    def __init__(
+        self,
+        image_size: Union[int, Tuple[int, int]] = 224,
+        num_channels: int = 3,
+        num_encoder_blocks: int = 4,
+        depths: List[int] = [2, 2, 2, 2],
+        sr_ratios: List[int] = [8, 4, 2, 1],
+        hidden_sizes: List[int] = [32, 64, 160, 256],
+        patch_sizes: List[int] = [7, 3, 3, 3],
+        strides: List[int] = [4, 2, 2, 2],
+        num_attention_heads: List[int] = [1, 2, 5, 8],
+        mlp_ratios: List[int] = [8, 8, 4, 4],
+        hidden_act: Union[str, Callable] = "gelu",
+        hidden_dropout_prob: float = 0.0,
+        attention_probs_dropout_prob: float = 0.0,
+        initializer_range: float = 0.02,
+        drop_path_rate: float = 0.0,
+        layer_norm_eps: float = 1e-6,
+        qkv_bias: bool = True,
+        linear_attention: bool = False,
+        out_features=None,
+        out_indices=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        if kwargs.get("_out_indices", None) is not None:
+            out_indices = kwargs["_out_indices"]
+            out_features = None
```

Comment:
So the explanation for this is that without either `out_indices` or `out_features` not being set to `None`, the call to `get_aligned_output_features_output_indices` on line 165 will end up overriding whatever is in the pretrained config with a single value representing the last stage, leading to loading issues of pretrained models expecting these values to be consistent.

However, this solution is not optimal because it will always favor the config over the user input, and we would prefer the user to have the option to override the config, so I've come up with a better solution, which will prioritize passing these values as 1) User setting if not None, 2) Config setting if available, 3) Default to None

I will have a commit soon to fix this

Note: I originally followed what I saw in the ResNet config, so this might still be a problem for loading models with ResNet backbones that had specific out_indices configured, we may want to test for that.

Reply:
I don't think this is in the ResNetConfig? 

It should be tested, but I believe is covered in `BackboneTesterMixin`

Reply:
Just to make sure we're on the same page - what do you mean by "overwrite the settings in the pretrained config"?

Reply:
When loading a pretrained config using `PvtV2Config.from_pretrained()`, the call to `super.__init__()` correctly finds and loads the `self._out_indices` and `self._out_features` from the JSON, and these are then accessible using the `self.out_indices` and `self.out_features` properties from the `BackboneConfigMixin`. However, the subsequent call to the `get_aligned_output_features_output_indices` function, if passed `None` for both `out_features` and `out_indices` (which are the default values in the `__init__` function and therefore the values when using `from_pretrained`), will actually revert these back to referencing only the last index/feature layer. Thus, the PVTv2Config, and presumably the ResNet config, would effectively *ignore* the `_out_indices` and `_out_features` fields from the pretrained JSON config. 

The code you originally highlighted was a hack to make sure that if there was already a self._out_indices set, that one of these variables would be not None when it reached the function call, but that was not the right solution. I believe the correct solution is to prioritize the values being fed to the function call as I laid out above.

I assumed that calling this line in the ResNetConfig `__init__` function was essential for some reason, which is why I copied it over, but then it caused problems. I think it's intended to allow for flexibility in loading different architectures, but I'm really not sure tbh, though I would hesitate to remove it.

Reply:
Actually on second thought, I can see that the call to the function is necessary for establishing a default behavior when the options are not passed, and also for aligning any customization that the user is trying to accomplish. The issue was that it did not account for previously loaded values from the JSON and was causing them to be overridden.

Reply:
> The issue was that it did not account for previously loaded values from the JSON and was causing them to be overridden.

Sorry, I'm still slightly confused about the behaviour being described here. Is this the preprocessor_config.json you're talking about? 

In this case, it should be possible to load previous values from the config i.e.

```py
config = PvtV2Config(out_indices=(2, 3))
config.save_pretrained("test_config")

config = PvtV2Config.from_pretrained("test_config")
assert config.out_indices == [2, 3]
```

Reply:
Yes, this is the behavior in question.

I've just run a test and the ResNet config works fine, it appears that there is a different saving behavior, where the ResNetConfig is saving these fields into the JSON as "out_indices" and "out_features", whereas the PvtV2Config is saving them with the underscore prefix as "_out_indices" and "_out_features", and this is where the divergent behavior stems from.

When I get time later this evening, I can dig into why `save_pretrained` has a different behavior on these fields in ResNetConfig vs PvtV2Config, because that is where the fix should probably be.